flags.h

```diff
@@ -34,6 +34,7 @@ struct Flags {
   bool is_dry_run;
   bool is_silent_mode;
   bool is_syntax_check_only;
+  bool posix_shell;
```

Comment:
This global state should be handled at evaluation time and it seems $(shell) should use this. I've moved this from here to eval.h. https://github.com/google/kati/commit/2941ea07651737a6151b19db45c010d2083dfa1d